ccl/changefeedccl: TestWebhookSink failed

ccl/changefeedccl.TestWebhookSink [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9868739?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9868739?buildTab=artifacts#/) on master @ [4619767bb0b75ac85ddcb76e33c73211e369afed](https://github.com/cockroachdb/cockroach/commits/4619767bb0b75ac85ddcb76e33c73211e369afed):


```
=== RUN   TestWebhookSink
    sink_webhook_test.go:108: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/sink_webhook_test.go:108
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/sink_webhook_test.go:164
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/sink_webhook_test.go:252
        	Error:      	Not equal: 
        	            	expected: int(0)
        	            	actual  : int64(1)
        	Test:       	TestWebhookSink
--- FAIL: TestWebhookSink (1.36s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101521 ccl/changefeedccl: TestWebhookSink failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWebhookSink.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27570

Epic CRDB-11732

--- a/pkg/ccl/changefeedccl/sink_webhook_test.go
+++ b/pkg/ccl/changefeedccl/sink_webhook_test.go
@@ -106,7 +106,13 @@ func testSendAndReceiveRows(t *testing.T, sinkSrc Sink, sinkDest *cdctest.MockWe
 	require.NoError(t, sinkSrc.EmitRow(ctx, nil, []byte("[1001]"), []byte("{\"after\":{\"col1\":\"val1\",\"rowid\":1000},\"key\":[1001],\"topic:\":\"foo\"}"), zeroTS, zeroTS, pool.alloc()))
 	require.NoError(t, sinkSrc.EmitRow(ctx, nil, []byte("[1001]"), []byte("{\"after\":{\"col1\":\"val1\",\"rowid\":1002},\"key\":[1001],\"topic:\":\"foo\"}"), zeroTS, zeroTS, pool.alloc()))
 	require.NoError(t, sinkSrc.Flush(ctx))
-	require.EqualValues(t, 0, pool.used())
+	testutils.SucceedsSoon(t, func() error {
+		remaining := pool.used()
+		if remaining == 0 {
+			return nil
+		}
+		return errors.Newf("waiting for 0 allocs (%d)", remaining)
+	})
 
 	require.Equal(t,
 		"{\"payload\":[{\"after\":{\"col1\":\"val1\",\"rowid\":1002},\"key\":[1001],\"topic:\":\"foo\"}],\"length\":1}", sinkDest.Latest(),

